app/src/main/java/com/kickstarter/viewmodels/ViewPledgeViewModel.java

```diff
@@ -248,8 +255,18 @@ public ViewModel(final @NonNull Environment environment) {
         .compose(bindToLifecycle())
         .subscribe(this.shippingSectionIsGone);
 
-      this.viewMessagesButtonIsGone = this.currentConfig.observable()
-        .map(config -> !ObjectUtils.coalesce(config.features().get(FeatureKey.ANDROID_MESSAGES), false));
+      Observable.zip(
+        isFromMessagesActivity,
+        featureFlagIsEnabled,
+        Pair::create
+      )
+        .map(fromMessagesAndFlagEnabled -> fromMessagesAndFlagEnabled.first || !fromMessagesAndFlagEnabled.second)
+        .compose(bindToLifecycle())
+        .subscribe(this.viewMessagesButtonIsGone::onNext);
```

Comment:
just curious why we're switching from an `Observable` to a `BehaviorSubject` here and below?